app/src/org/commcare/android/database/user/UserDatabaseUpgrader.java

```diff
@@ -208,15 +221,71 @@ private boolean upgradeSevenEight(SQLiteDatabase db, int oldVersion, int newVers
         }
     }
 
-    /**
+    /*
      * Deserialize user fixtures in db using old form instance serialization
      * scheme, and re-serialize them using the new scheme that preserves
      * attributes.
      */
-    private boolean upgradeEightNine(SQLiteDatabase db, int oldVersion, int newVersion) {
+    private boolean upgradeEightNine(SQLiteDatabase db) {
         return FixtureSerializationMigration.migrateFixtureDbBytes(db, c);
     }
 
+    /**
+     * Adding an appId field to FormRecords, for compatibility with multiple apps functionality
+     */
+    private boolean upgradeNineTen(SQLiteDatabase db) {
+        db.beginTransaction();
+        try {
+            if (multipleInstalledAppRecords()) {
+                // Cannot migrate FormRecords once this device has already started installing
+                // multiple applications, because there is no way to know which of those apps the
+                // existing FormRecords belong to
+                throw new MigrationException(true);
+            }
+
+            SqlStorage<FormRecordV1> oldStorage = new SqlStorage<>(
+                    FormRecord.STORAGE_KEY,
+                    FormRecordV1.class,
+                    new ConcreteAndroidDbHelper(c, db));
+
+            String appId = getInstalledAppRecord().getApplicationId();
+            Vector<FormRecord> upgradedRecords = new Vector<>();
+            // Create all of the updated records, based upon the existing ones
+            for (FormRecordV1 oldRecord : oldStorage) {
+                FormRecord newRecord = new FormRecord(
+                        oldRecord.getInstanceURIString(),
+                        oldRecord.getStatus(),
+                        oldRecord.getFormNamespace(),
+                        oldRecord.getAesKey(),
+                        oldRecord.getInstanceID(),
+                        oldRecord.lastModified(),
+                        appId);
+                newRecord.setID(oldRecord.getID());
+                upgradedRecords.add(newRecord);
+            }
+
+            // Alter the FormRecord table to include an app id column
+            db.execSQL(DatabaseAppOpenHelper.addColumnToTable(
+                    FormRecord.STORAGE_KEY,
+                    FormRecord.META_APP_ID,
```

Comment:
Ok, yeah, should definitely be sanitizing either here or inside the addColumnToTable method